pkg/driver/controller_server.go

```diff
@@ -457,6 +455,32 @@ func (driver *Driver) createVolume(
 				return nil, status.Error(codes.Internal, "Parent volume ID is missing in the snapshot response")
 			}
 
+			// Get the parent volume from the snapshot
+			existingParentVolume, err := storageProvider.GetVolume(existingSnap.VolumeID)
+			if err != nil {
+				log.Error("err: ", err.Error())
+				return nil,
+					status.Error(codes.Internal,
+						fmt.Sprintf("Failed to check if snapshot's parent volume %s with ID %s exist, err: %s",
+							existingSnap.VolumeName, existingSnap.VolumeID, err.Error()))
+			}
+
+			// Get existing volume device info
+			parentVolFsType, err := driver.flavor.GetVolumePropertyOfPV("fsType", existingParentVolume.Name)
```

Comment:
Please add a nil check for existingParentVolume here. When not found, this will be nil along with err=nil.

Reply:
existingParentVolume nil check is handled above . Not sure if you meant that.

Reply:
existingParentVolume nil check is handled above . Not sure if you meant that.

Reply:
Where is it handled? i don't see this after line 459:

Reply:
Please check the updated code. thx

Reply:
looks good. Thanks.